### ISSUE TITLE
implement polylog for arbitrarily large arguments.

### ISSUE THREAD

**Issue by lemmatum** [fredrik-johansson/mpmath#390]
We've been running into an [issue](https://github.com/PlasmaPy/PlasmaPy/issues/245) over at [PlasmaPy](https://github.com/PlasmaPy/PlasmaPy), where when we try to use `polylog` to calculate the [Fermi Integral](https://en.wikipedia.org/wiki/Complete_Fermi%E2%80%93Dirac_integral) we hit an implementation error:

```
import numpy as np
import mpmath
mpmath.polylog(1 + 0.5, -np.exp(3.89))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/anaconda3/lib/python3.6/site-packages/mpmath/ctx_mp_python.py", line 1016, in f_wrapped
    retval = f(ctx, *args, **kwargs)
  File "/home/user/anaconda3/lib/python3.6/site-packages/mpmath/functions/zeta.py", line 482, in polylog
    return polylog_general(ctx, s, z)
  File "/home/user/anaconda3/lib/python3.6/site-packages/mpmath/functions/zeta.py", line 449, in polylog_general
    raise NotImplementedError("polylog for arbitrary s and z")
NotImplementedError: polylog for arbitrary s and z

```

Would it be possible to implement `polylog` for large arguments? For our purposes it is fine to restrict it to an index of 1.5 for now.

**Comment by fredrik-johansson** [comment]
Yes, it would be possible. The decomposition in terms of the Hurwitz zeta function could be used for this. Here is code that should work at least in your case:

    def polylogb(s,z):
        s = mpmathify(s)
        z = mpmathify(z)
        v = 1-s
        y = log(-z)/(2*pi*j)
        return gamma(v)*(1j**v*zeta(v,0.5+y) + 1j**-v*zeta(v,0.5-y))/(2*pi)**v

    >>> polylogb(1+0.5, -exp(3.89))
    mpc(real='-6.2729922293118188', imag='0.0')

Beware that there could be some ranges where this formula loses precision.

**Comment by lemmatum** [comment]
@fredrik-johansson would such a code be appropriate within the `mpmath` package? In particular, would you like me to create a PR for extending `polylog` with this decomposition?

**Comment by fredrik-johansson** [comment]
I have fixed the bug in zeta. In fact, a part of the code was using polylog indirectly, which not only seems to have been incorrect on branch cuts in some cases; it would also have made it circular to use polylog from zeta...

**Comment by fredrik-johansson** [comment]
Fixed in f90e0df22e6e578164f737964ff8ae677c2a537b.

The problem remains that polylog does not use the best algorithm in all cases, but not raising an exception here should be an improvement.

@grozin I don't know, but I don't really think it would help. The standard algorithms for the usual polylogarithm are already quite efficient (just not implemented very well in mpmath at the moment).
